thrift_rpc/recorder.go

```diff
@@ -385,6 +387,19 @@ func (r *Recorder) Flush() {
 	}
 }
 
+func (r *Recorder) Close() error {
+	r.lock.Lock()
+	closech := r.closech
+	r.closech = nil
+	r.lock.Unlock()
+
+	if closech != nil {
+		close(closech)
+	}
+
+	return nil
```

Comment:
I agree this is a race, but as I mentioned in the commit message, this PR does not attempt to make any improvements to the thrift tracer. The thrift tracer is currently scheduled for deletion. If, for some reason, the thrift tracer must have a renewed lease on life, we will improve it in a separate PR, as there are several issues with it. This code simply attempts to not break the the thrift tracer in the process of adding this feature.